hab-pkg-deb/bin/hab-pkg-deb.sh

```diff
@@ -0,0 +1,390 @@
+  du "$deb_context" --apparent-size --block-size=1024 --summarize | cut -f1
+}
+
+# The package priority.
+#
+# Can be one of required, important, standard, optional, or extra.
+# See https://www.debian.org/doc/manuals/debian-faq/ch-pkg_basics.en.html#s-priority
+#
+# TODO: Allow customizing this
+priority() {
+  echo extra
+}
+
+# The package section.
+#
+# See https://www.debian.org/doc/debian-policy/ch-archive.html#s-subsections
+#
+# TODO: Allow customizing this
+section() {
+  echo misc
+}
+
+# parse the CLI flags and options
+parse_options() {
+  echo "@=$@"
+  opts="$(getopt \
+    --longoptions help,version,preinst:,postinst:,prerm:,postrm:,conflicts:,provides:,replaces: \
+    --name "$program" --options h,V -- "$@" \
+  )"
+  eval set -- "$opts"
+
+  while :; do
+    case "$1" in
+      -h | --help)
+        print_help
+        exit
+        ;;
+      -V | --version)
+        echo "$program $version"
+        exit
+        ;;
+      --preinst)
+        preinst=$2
+        shift 2
+        ;;
+      --postinst)
+        postinst=$2
+        shift 2
+        ;;
+      --prerm)
+        prerm=$2;
+        shift 2
+        ;;
+      --postrm)
+        postrm=$2
+        shift 2
+        ;;
+      --conflicts)
+        conflicts=$2
+        shift 2
+        ;;
+      --provides)
+        provides=$2
+        shift 2
+        ;;
+      --replaces)
+        replaces=$2
+        shift 2
+        ;;
+      --)
+        shift
+        pkg=$*
+        break
+        ;;
+      *)
+        exit_with "Unknown error" 1
+        ;;
+    esac
+  done
+  if [ -z "$pkg" ] || [ "$pkg" = "--" ]; then
+    print_help
+    exit_with "You must specify a Habitat package." 1
+  fi
+}
+
+# The name converted to all lowercase to be compatible with Debian naming
+# conventions
+convert_name() {
+  safe_name="${pkg_origin,,}-${pkg_name,,}"
+}
+
+# Return the Debian-ready version, replacing all dashes (-) with tildes
+# (~) and converting any invalid characters to underscores (_).
+convert_version() {
+  if [[ $pkg_version == *"-"* ]]; then
+    safe_version="${pkg_version//-/\~}"
+    warn "Dashes hold special significance in the Debian package versions. "
+    warn "Versions that contain a dash and should be considered an earlier "
+    warn "version (e.g. pre-releases) may actually be ordered as later "
+    warn "(e.g. 12.0.0-rc.6 > 12.0.0). We'll work around this by replacing "
+    warn "dashes (-) with tildes (~). Converting '$pkg_version' "
+    warn "to '$safe_version'."
+	else
+    safe_version="$pkg_version"
+	fi
+}
+
+# Output the contents of the "control" file
+render_control_file() {
+# TODO: Depends/conflicts/provides, etc. See https://www.debian.org/doc/debian-policy/ch-relationships.html
+# TODO: Should vendor be the origin or not?
+control=$(cat <<EOF
+Package: $safe_name
+Version: $safe_version-$pkg_release
+Vendor: $pkg_origin
+Architecture: $(architecture)
+Installed-Size: $(installed_size)
+Section: $(section)
+Priority: $(priority)
+EOF
+)
```

Comment:
Can we render the control files off of a Handlebars template (same templating syntax the rest of Habitat uses). Since we control the deps for all of our exporters we could leverage a CLI (like [handlebars-cmd](https://github.com/spion/handlebars-cmd) or [hbs-cli](https://github.com/keithamus/hbs-cli) if we created a [core-plan](https://github.com/habitat-sh/core-plans) for it.

This approach would make all of the control file rendering code much more readable and maintainable. It also opens up the ability for an end-user to provide their own control file template.